cms/src/components/groups/GroupSelector.spec.ts

```diff
@@ -0,0 +1,173 @@
+    accessMap.value = superAdminAccessMap;
+
+    beforeAll(async () => {
+        await db.docs.bulkPut([
+            mockGroupDtoPublicContent,
+            mockGroupDtoPublicEditors,
+            mockGroupDtoPublicUsers,
+            mockGroupDtoPrivateContent,
+        ]);
+    });
+
+    beforeEach(async () => {
+        // setActivePinia(createTestingPinia());
+    });
+
+    afterEach(async () => {
+        vi.clearAllMocks();
+    });
+
+    afterAll(async () => {
+        await db.docs.clear();
+        await db.localChanges.clear();
+    });
+
+    it("displays selected groups", async () => {
+        const wrapper = mount(GroupSelector, {
+            props: {
+                groups: [mockGroupDtoPublicContent._id],
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.text()).not.toContain("Private Content");
+            expect(wrapper.text()).toContain("Public Content");
+        });
+    });
+
+    it("displays all available groups", async () => {
+        const wrapper = mount(GroupSelector, {
+            props: {
+                groups: [],
+            },
+        });
+
+        await wrapper.find("button").trigger("click"); // First button is the dropdown button
+
+        await waitForExpect(() => {
+            expect(wrapper.text()).toContain("Public Content");
+            expect(wrapper.text()).toContain("Public Users");
+            expect(wrapper.text()).toContain("Public Editors");
+            expect(wrapper.text()).toContain("Private Content");
+        });
+    });
+
+    it("hides groups to which the user does not have assign access", async () => {
+        // Remove assign access to the Public Users group
+        // @ts-ignore
+        accessMap.value[mockGroupDtoPublicUsers._id].group.assign = false;
+
+        const wrapper = mount(GroupSelector, {
+            props: {
+                groups: [],
+            },
+        });
+
+        await wrapper.find("button").trigger("click"); // First button is the dropdown button
+
+        await waitForExpect(() => {
+            expect(wrapper.text()).toContain("Public Content");
+            expect(wrapper.text()).toContain("Public Editors");
+            expect(wrapper.text()).toContain("Private Content");
+
+            expect(wrapper.text()).not.toContain("Public Users");
+        });
+    });
+
+    it("can filter on groups", async () => {
+        const wrapper = mount(GroupSelector, {
+            props: {
+                groups: [],
+            },
+        });
+
+        await wrapper.find("input").setValue("edit");
+
+        await waitForExpect(() => {
+            expect(wrapper.text()).toContain("Public Editors");
+            expect(wrapper.text()).not.toContain("Public Content");
+            expect(wrapper.text()).not.toContain("Super Admins");
+        });
+    });
+
+    it.skip("Updates the passed array when selecting a group", async () => {
```

Comment:
Please add a note with a link to the relevant issue why this test is skipped so we don't lose track of it.

Reply:
Issue #254 added with reference to this PR